Build failure: linuxPackages_testing.broadcom-sta

### Nixpkgs version

Please select a version.

### Steps to reproduce

Build linuxPackages_testing  (linux 6.14) with broadcom-sta and the wl kernel module

### Can Hydra reproduce this build failure?

No, Hydra cannot reproduce this build failure.

### Link to Hydra build job

_No response_

### Relevant log output

```console
brush$ sudo nixos-rebuild switch --flake .#macbookair71 --upgrade
warning: Git tree '/etc/nixos' is dirty
building the system configuration...
warning: Git tree '/etc/nixos' is dirty
error: builder for '/nix/store/3iync2n1kdkbhrkw6ac8lz34vrmp4hhc-broadcom-sta-6.30.223.271-6.14.drv' failed with exit code 2;
       last 25 log lines:
       >                  from /nix/store/1v70gd12v5chh69vqb0ld3xaqaz298sn-linux-6.14-dev/lib/modules/6.14.0/source/include/linux/umh.h:4,
       >                  from /nix/store/1v70gd12v5chh69vqb0ld3xaqaz298sn-linux-6.14-dev/lib/modules/6.14.0/source/include/linux/kmod.h:9,
       >                  from /nix/store/1v70gd12v5chh69vqb0ld3xaqaz298sn-linux-6.14-dev/lib/modules/6.14.0/source/include/linux/module.h:17,
       >                  from ././src/include/linuxver.h:40,
       >                  from src/wl/sys/wl_linux.c:27:
       > src/wl/sys/wl_linux.c: In function 'wl_down':
       > /nix/store/1v70gd12v5chh69vqb0ld3xaqaz298sn-linux-6.14-dev/lib/modules/6.14.0/source/include/linux/workqueue.h:765:9: warning: call to '__warn_flushing_systemwide_wq' declared with attribute warning: Please avoid flushing system-wide workqueues. []
       >   765 |         __warn_flushing_systemwide_wq();                                \
       >       |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
       > src/wl/sys/wl_linux.c:1499:25: note: in expansion of macro 'flush_scheduled_work'
       >  1499 |                         flush_scheduled_work();
       >       |                         ^~~~~~~~~~~~~~~~~~~~
       >   CC [M]  src/wl/sys/wl_iw.o
       >   CC [M]  src/wl/sys/wl_cfg80211_hybrid.o
       > src/wl/sys/wl_cfg80211_hybrid.c:1898:25: error: initialization of 'int (*)(struct wiphy *, struct wireless_dev *, unsigned int,  int *)' from incompatible pointer type 's32 (*)(struct wiphy *, struct wireless_dev *, s32 *)' {aka 'int (*)(struct wiphy *, struct wireless_dev *, int *)'} []
       >  1898 |         .get_tx_power = wl_cfg80211_get_tx_power,
       >       |                         ^~~~~~~~~~~~~~~~~~~~~~~~
       > src/wl/sys/wl_cfg80211_hybrid.c:1898:25: note: (near initialization for 'wl_cfg80211_ops.get_tx_power')
       > make[4]: *** [/nix/store/1v70gd12v5chh69vqb0ld3xaqaz298sn-linux-6.14-dev/lib/modules/6.14.0/source/scripts/Makefile.build:207: src/wl/sys/wl_cfg80211_hybrid.o] Error 1
       > make[3]: *** [/nix/store/1v70gd12v5chh69vqb0ld3xaqaz298sn-linux-6.14-dev/lib/modules/6.14.0/source/Makefile:1994: .] Error 2
       > make[2]: *** [/nix/store/1v70gd12v5chh69vqb0ld3xaqaz298sn-linux-6.14-dev/lib/modules/6.14.0/source/Makefile:251: __sub-make] Error 2
       > make[2]: Leaving directory '/build/broadcom-sta'
       > make[1]: *** [/nix/store/1v70gd12v5chh69vqb0ld3xaqaz298sn-linux-6.14-dev/lib/modules/6.14.0/source/Makefile:251: __sub-make] Error 2
       > make[1]: Leaving directory '/nix/store/1v70gd12v5chh69vqb0ld3xaqaz298sn-linux-6.14-dev/lib/modules/6.14.0/build'
       > make: *** [Makefile:173: all] Error 2
       For full logs, run 'nix log /nix/store/3iync2n1kdkbhrkw6ac8lz34vrmp4hhc-broadcom-sta-6.30.223.271-6.14.drv'.
error: 1 dependencies of derivation '/nix/store/ihnndb83vyr6qbbc9dml4jvg4mbh88dm-linux-6.14-modules.drv' failed to build
error: 1 dependencies of derivation '/nix/store/2fcrdmcc8917cjcn2n938l6hnnm9shb8-nixos-system-nixos-25.05.20250325.33319c3.drv' failed to buil
```

### Additional context

I'm not sure if hydra can reproduce this build failure as the search bar gives a database timeout error: 
```
DBIx::Class::Storage::DBI::_dbh_execute(): DBI Exception: DBD::Pg::st execute failed: ERROR: canceling statement due to statement timeout [for Statement "SELECT me.id, me.finished, me.timestamp, me.jobset_id, me.job, me.nixname, me.description, me.drvpath, me.system, me.license, me.homepage, me.maintainers, me.maxsilent, me.timeout, me.ischannel, me.iscurrent, me.priority, me.globalpriority, me.starttime, me.stoptime, me.iscachedbuild, me.buildstatus, me.size, me.closuresize, me.releasename, me.keep, me.notificationpendingsince FROM builds me LEFT JOIN buildoutputs buildoutputs ON buildoutputs.build = me.id WHERE ( buildoutputs.path ILIKE ? ) ORDER BY id desc LIMIT ?" with ParamValues: 1='%broadcom-sta%', 2='10'] at /nix/store/dcdf80y3y6f1pa4l3r1y0asmy04bgl6m-hydra-0.1.20250216.c3b6e7b/libexec/hydra/lib/Hydra/Controller/Root.pm line 518
```

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.13.5, NixOS, 25.05 (Warbler), 25.05.20250325.33319c3`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.12`
 - nixpkgs: `/nix/store/vsdijys7qrif2rg7i5cb63yqf6k9vgp7-source`


### Notify maintainers

@j0hax 

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/os-specific/linux/broadcom-sta/default.nix
+++ b/pkgs/os-specific/linux/broadcom-sta/default.nix
@@ -96,7 +96,10 @@ stdenv.mkDerivation {
     description = "Kernel module driver for some Broadcom's wireless cards";
     homepage = "http://www.broadcom.com/support/802.11/linux_sta.php";
     license = lib.licenses.unfreeRedistributable;
-    maintainers = [ lib.maintainers.j0hax ];
+    maintainers = with lib.maintainers; [
+      j0hax
+      nullcube
+    ];
     platforms = lib.platforms.linux;
   };
 }

--- a/pkgs/os-specific/linux/broadcom-sta/default.nix
+++ b/pkgs/os-specific/linux/broadcom-sta/default.nix
@@ -111,5 +111,15 @@ stdenv.mkDerivation {
       "i686-linux"
       "x86_64-linux"
     ];
+    knownVulnerabilities = [
+      "CVE-2019-9501: heap buffer overflow, potentially allowing remote code execution by sending specially-crafted WiFi packets"
+      "CVE-2019-9502: heap buffer overflow, potentially allowing remote code execution by sending specially-crafted WiFi packets"
+      (
+        "The Broadcom STA wireless driver is not maintained "
+        + "and is incompatible with Linux kernel security mitigations. "
+        + "It is heavily recommended to replace the hardware and remove the driver. "
+        + "Proceed at your own risk!"
+      )
+    ];
   };
 }

--- a/pkgs/os-specific/linux/broadcom-sta/default.nix
+++ b/pkgs/os-specific/linux/broadcom-sta/default.nix
@@ -94,12 +94,15 @@ stdenv.mkDerivation {
 
   meta = {
     description = "Kernel module driver for some Broadcom's wireless cards";
-    homepage = "http://www.broadcom.com/support/802.11/linux_sta.php";
+    homepage = "https://www.broadcom.com/support/download-search?pg=Legacy%20Products&pf=Legacy%20Wireless&pn&pa&po&dk&pl";
     license = lib.licenses.unfreeRedistributable;
     maintainers = with lib.maintainers; [
       j0hax
       nullcube
     ];
-    platforms = lib.platforms.linux;
+    platforms = [
+      "i686-linux"
+      "x86_64-linux"
+    ];
   };
 }

--- a/pkgs/os-specific/linux/broadcom-sta/default.nix
+++ b/pkgs/os-specific/linux/broadcom-sta/default.nix
@@ -3,12 +3,15 @@
   stdenv,
   fetchurl,
   fetchFromGitHub,
-  fetchpatch2,
   kernel,
 }:
 
 let
   version = "6.30.223.271";
+  # Patchset release number from rpmfusion, to more easily differentiate
+  # versions and updates. See `wl-kmod.spec` file:
+  # https://github.com/rpmfusion/wl-kmod/blob/master/wl-kmod.spec#L19
+  release = "57";
   hashes = {
     i686-linux = "sha256-T4twspOsjMXHDlca1dGHjQ8p0TOkb+eGmGjZwZtQWM0=";
     x86_64-linux = "sha256-X3l3TVvuyPdja1nA+wegMQju8eP9MkVjiyCFjHFBRL4=";
@@ -21,8 +24,8 @@ let
   rpmFusionPatches = fetchFromGitHub {
     owner = "rpmfusion";
     repo = "wl-kmod";
-    rev = "9a5a0d7195e0f6b05ff97e948b97fb0b7427cbf2";
-    hash = "sha256-pOOkkOjc77KGqc9fWuRyRsymd90OpLEnbOvxBbeIdKQ=";
+    rev = "b0d19578ebd0daae9c5b7f9e9511a6d73ac4d957";
+    hash = "sha256-v7mZ2S/eVfGTEXrxpdiemHhrC+P3/sPOZqTBhRtins4=";
   };
   patchset = [
     "wl-kmod-001_wext_workaround.patch"
@@ -55,10 +58,14 @@ let
     "wl-kmod-028_kernel_6.12_adaptation.patch"
     "wl-kmod-029_kernel_6.13_adaptation.patch"
     "wl-kmod-030_kernel_6.14_adaptation.patch"
+    "wl-kmod-031_replace_EXTRA_CFLAGS_EXTRA_LDFLAGS_with_ccflags-y_ldflags-y.patch"
+    "wl-kmod-032_add_MODULE_DESCRIPTION_macro.patch"
+    "wl-kmod-033_disable_objtool_add_warning_unmaintained.patch"
+    "wl-kmod-034_kernel_6.15_adaptation_replace_del_timer_with_timer_delete.patch"
   ];
 in
 stdenv.mkDerivation {
-  name = "broadcom-sta-${version}-${kernel.version}";
+  name = "broadcom-sta-${version}-${release}-${kernel.version}";
 
   src = fetchurl {
     url = "https://docs.broadcom.com/docs-and-downloads/docs/linux_sta/${tarball}";

